exec/run_test.go

```diff
@@ -41,3 +42,48 @@ func TestDefaultEnv(t *testing.T) {
 	})
 	assert.Contains(t, c.Env, "alice=bob")
 }
+
+func TestRun(t *testing.T) {
+	command := exec.Cmd{
+		Name: "pwd",
+	}
+	stdout, _, err := exec.Run(command)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, stdout)
+}
+
+func TestRunTimeout(t *testing.T) {
+	command := exec.Cmd{
+		Name:    "sleep",
+		Argv:    []string{"2"},
+		Timeout: 3 * time.Second,
+	}
+	_, _, err := exec.Run(command)
+	assert.NoError(t, err)
+
+	command.Timeout = 1 * time.Second
+	_, _, err = exec.Run(command)
+	assert.Contains(t, err.Error(), "timed out")
+}
+
+func TestRunRetry(t *testing.T) {
+	command := exec.Cmd{
+		Name:    "sleep",
+		Argv:    []string{"3"},
+		Timeout: 1 * time.Second,
+		Retries: 1,
+	}
+
+	start := time.Now()
+	_, _, err := exec.Run(command)
+	assert.Error(t, err)
+	assert.WithinDuration(t, start, time.Now(), 3*time.Second)
+
+	command.Retries = 4
+	start = time.Now()
+	_, _, err = exec.Run(command)
```

Comment:
the assertions that follow on lines 87 and 88 rely on the time when `time.Since` is invoked. Instead, we should mark the time immediately after `exec.Run` returns and compare the end time to the start time:
```
start = time.Now()
_, _, err = exec.Run(command)
end := time.Now()
assert.Error(t, err)
// 4 retries with a 1 second timeout means that the command should take 5 seconds to error.
assert.True(t, end.Sub(start) > 5*time.Second)
assert.True(t, end.Sub(start) < 6*time.Second)
```

Reply:
this also makes it a little more obvious what is being tested